app/shop.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from typing import List, Any
+import datetime
+from app.json_reader import data, fuel_price
+from app.car import Car
+from app.customer import Customer
+
+
+@dataclass
+class Shop:
+    name: str
+    location: List[int]
+    products: dict
+
+    def __hash__(self) -> int:
+        return hash(self.name)
+
+    @staticmethod
+    def create_shops() -> list:
+        return [
+            Shop(
+                name=shop["name"],
+                location=shop["location"],
+                products=shop["products"]
+            )
+            for shop in data["shops"]
+        ]
+
+    def calculate_shop_trip(self, customer: Customer) -> dict:
+        total_cost = sum(
+            [
+                round(
+                    (
+                        customer.product_cart[product] * self.products[product]
+                    ), 2
+                )
+                for product in customer.product_cart
+            ]
+        )
+        fuel_cost = Car.calculate_fuel_cost(
+            customer.location,
+            self.location,
+            fuel_price,
+            customer.car["fuel_consumption"]
+        )
+        total_cost_per_shop = {self: round(total_cost + fuel_cost, 2)}
+        return total_cost_per_shop
+
+    @staticmethod
+    def find_cheapest_shop(shops: list, customer: Customer) -> Any:

```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]

invalid return typehint